src/Compatibility/Microsoft.DotNet.ApiCompatibility/Runner/ApiCompatRunner.cs

```diff
@@ -0,0 +1,173 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.CodeAnalysis;
+using Microsoft.DotNet.ApiCompatibility.Abstractions;
+using Microsoft.DotNet.ApiCompatibility.Logging;
+
+namespace Microsoft.DotNet.ApiCompatibility.Runner
```

Comment:
Was anything changed in these files other than the interface abstraction? It's hard when the diff shows the whole files as this essentially means everything in compat must be re-reviewed.

Reply:
The ones that I already touched on in previous meetings:
- Removed package specific log to make it usable outside of PackageValidation.
- Dependencies are moved into the constructor so that the type becomes unit testable (by mocking the external dependencies).

In addition to that:
- More diagnostics with low importance logged so that they appear in a binlog / high verbosity text log
- Strongly type work items and store them in a `HashSet<T>` instead of a `Dictionary<T,U>`.